spec/features/admin_can_bulk_deactivate_suppliers_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
+
+RSpec.feature 'Admin can bulk deactivate suppliers' do
+  let!(:supplier) do
+    supplier = create(:supplier,
```

Comment:
I think you can remove the `supplier` variable here, as the factory will return itself.

Reply:
Have actually removed the setting of the `agreement` as a separate `let` below and moved it into here now.